From 5df5ef4f26dfc7f5e5c55b96045c5278a372e8dc Mon Sep 17 00:00:00 2001
From: Cristian Stoica <cristian.stoica@freescale.com>
Date: Thu, 5 Dec 2013 14:34:57 +0200
Subject: [PATCH] workaround for ENG00289183

- this disables support for algorithms that may not be activated in
  kernel to avoid a crash during cryptodev session create

Change-Id: Ibd50b1afb7404360cd89a6606b21bb0105af6994
Signed-off-by: Cristian Stoica <cristian.stoica@freescale.com>
---
 crypto/engine/eng_cryptodev.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/crypto/engine/eng_cryptodev.c b/crypto/engine/eng_cryptodev.c
index 389514f..ad123ee 100644
--- a/crypto/engine/eng_cryptodev.c
+++ b/crypto/engine/eng_cryptodev.c
@@ -145,15 +145,15 @@ static struct {
 	int	keylen;
 	int	mackeylen;
 } ciphers[] = {
-	{ CRYPTO_ARC4,          NID_rc4,          0,  16, 0},
+/*	{ CRYPTO_ARC4,          NID_rc4,          0,  16, 0},*/
 	{ CRYPTO_DES_CBC,       NID_des_cbc,	  8,  8,  0},
 	{ CRYPTO_3DES_CBC,      NID_des_ede3_cbc, 8,  24, 0},
 	{ CRYPTO_AES_CBC,       NID_aes_128_cbc,  16, 16, 0},
 	{ CRYPTO_AES_CBC,       NID_aes_192_cbc,  16, 24, 0},
 	{ CRYPTO_AES_CBC,       NID_aes_256_cbc,  16, 32, 0},
-	{ CRYPTO_BLF_CBC,       NID_bf_cbc,       8,  16, 0},
+/*	{ CRYPTO_BLF_CBC,       NID_bf_cbc,       8,  16, 0},
 	{ CRYPTO_CAST_CBC,      NID_cast5_cbc,    8,  16, 0},
-	{ CRYPTO_SKIPJACK_CBC,  NID_undef,        0,  0,  0},
+	{ CRYPTO_SKIPJACK_CBC,  NID_undef,        0,  0,  0},*/
 	{ CRYPTO_TLS10_AES_CBC_HMAC_SHA1, NID_aes_128_cbc_hmac_sha1, 16, 16, 20},
 	{ 0, NID_undef,	0, 0, 0},
 };
@@ -166,9 +166,9 @@ static struct {
 } digests[] = {
 	{ CRYPTO_MD5_HMAC,		NID_hmacWithMD5,	16},
 	{ CRYPTO_SHA1_HMAC,		NID_hmacWithSHA1,	20},
-	{ CRYPTO_RIPEMD160_HMAC,	NID_ripemd160,		16/*?*/},
+/*	{ CRYPTO_RIPEMD160_HMAC,	NID_ripemd160,	16},
 	{ CRYPTO_MD5_KPDK,		NID_undef,		0},
-	{ CRYPTO_SHA1_KPDK,		NID_undef,		0},
+	{ CRYPTO_SHA1_KPDK,		NID_undef,		0},*/
 	{ CRYPTO_MD5,			NID_md5,		16},
 	{ CRYPTO_SHA1,			NID_sha1,		20},
 	{ 0,				NID_undef,		0},
-- 
1.7.11.7

